contentcuration/contentcuration/assets/channelList/vuex/channelList/mutations.js

```diff
@@ -0,0 +1,73 @@
+import Vue from 'vue';
+import { prepChannel, setDirtyTracking } from '../../utils';
+
+/* CHANNEL LIST MUTATIONS */
+function mergeChannel(channelsMap, channel) {
+  setDirtyTracking(channel);
+  return {
+    ...channelsMap,
+    [channel.id]: {
+      ...channelsMap[channel.id],
+      ...channel,
+    },
+  };
+}
+
+export function ADD_CHANNEL(state, channel) {
+  state.channelsMap = mergeChannel(state.channelsMap, channel);
+}
+
+export function ADD_CHANNELS(state, channels = []) {
+  state.channelsMap = channels.reduce((channelsMap, channel) => {
+    return mergeChannel(channelsMap, channel);
+  }, state.channelsMap);
+}
+
+export function REMOVE_CHANNEL(state, channelId) {
+  Vue.delete(state.channelsMap, channelId);
```

Comment:
This is a nice gotcha - just calling `delete` does not trigger vue reactivity.

Reply:
It is removing an object in the channelsMap - the values of the object are presumed to be objects that contain the data for a channel, so it has to be completely removed. See the `channels` getter below that returns an array of the values of this map.